opm/porsol/mimetic/IncompFlowSolverHybrid.hpp

```diff
@@ -1711,7 +1710,7 @@ namespace Opm {
 
                 std::transform(pi.begin(), pi.end(),
                                pi.begin(),
-                               boost::bind(std::minus<Scalar>(), p[c0], _1));
+                               [p, c0](const double& input) { return p[c0] - input; });
```

Comment:
Is capturing a copy of `p`, which is a vector, really a good idea? Not sure how big the vector is...

Reply:
> Not sure how big the vector is

Size equal to number of (active) cells.

Reply:
This should make the upscaling code O(n^2) then... I'll push a fix if nobody else already is on it?

Reply:
Strangely, this seems to not hurt. Perhaps the compiler optimizes it away. Even stranger, when trying to run a "good old testcase" that I have been using since 2007 or so, the linear solver iterations fails immediately, as if the matrix was NaN, but the AMG hierarchy building is fine. I'll add this to today's mysteries and try to address both.

Reply:
Depending on what `Scalar` is (e.g. `float`), the new code might not be equivalent. ``std::minus<Scalar>` always returned a `Scalar` and not a double.